app/models/name/propagate_generic_classifications.rb

```diff
@@ -44,20 +44,33 @@ def build_accepted_names_lookup_table
     end
 
     def accepted_generic_classification_strings
-      Name.where(rank: Name.ranks[:Genus], deprecated: false).
-        where("author NOT LIKE 'sensu lato%'").
-        where("LENGTH(classification) > 2").
-        pluck(:text_name, :classification).
+      select_manager = arel_select_accepted_generic_classification_strings
+      Name.connection.select_rows(select_manager.to_sql).
         each_with_object({}) do |vals, classifications|
-          text_name, classification = vals
-          if classifications[text_name].present?
-            warn("Multiple accepted non-sensu lato genera for #{text_name}!")
-          else
-            classifications[text_name] = classification
-          end
+        text_name, classification = vals
+        if classifications[text_name].present?
+          warn("Multiple accepted non-sensu lato genera for #{text_name}!")
+        else
+          classifications[text_name] = classification
         end
+      end
+    end
+
+    # Name.where(rank: Name.ranks[:Genus], deprecated: false).
+    #   where("author NOT LIKE 'sensu lato%'").
+    #   where("LENGTH(classification) > 2").
+    #   pluck(:text_name, :classification).
+    def arel_select_accepted_generic_classification_strings
+      names = Name.arel_table
+      names.where(names[:rank].eq(Name.ranks[:Genus]).
+        and(names[:deprecated].eq(Arel.sql("FALSE"))).
+        and(names[:author].does_not_match("sensu lato%")).
+        and(names[:classification].length > 2)).
+        project(names[:text_name], names[:classification])
```

Comment:
This is admittedly pretty clean already.  But in the spirit of using ActiveRecord wherever it is at least as concise and efficient as Arel, could we not also do this as:

`Name.where(rank: Name.ranks[:Genus], deprecated: false).`
`          where(Name[:author].does_not_match("sensu lato%")).`
`          where(Name[:classification[.length > 2)).`
`          pluck(:text_name, :classification).`
`          each_with_object({}) do ...`

This stays a single method, no helpers, and it is all db agnostic.